packages/styled-components/transforms/src/visitors/transpile_css_prop/top_level_binding_collector.rs

```diff
@@ -0,0 +1,82 @@
+use swc_common::collections::AHashSet;
+use swc_ecmascript::{
+    ast::{
+        ArrowExpr, ClassDecl, FnDecl, Function, ImportDefaultSpecifier, ImportNamedSpecifier,
+        ImportStarAsSpecifier, ObjectPatProp, Pat,
+    },
+    utils::{ident::IdentLike, Id},
+    visit::{noop_visit_type, Visit, VisitWith},
+};
+
+// Modified from swc_ecma_utils/src/lib.rs:BindingCollector.
+pub struct TopLevelBindingCollector {
+    bindings: AHashSet<Id>,
+}
+
+impl TopLevelBindingCollector {
+    fn add(&mut self, i: &Id) {
+        self.bindings.insert(i.clone());
+    }
+}
+
+impl Visit for TopLevelBindingCollector {
+    noop_visit_type!();
+
+    fn visit_class_decl(&mut self, node: &ClassDecl) {
+        self.add(&node.ident.to_id());
+    }
+
+    fn visit_fn_decl(&mut self, node: &FnDecl) {
+        self.add(&node.ident.to_id());
+    }
+
+    fn visit_pat(&mut self, node: &Pat) {
+        match node {
+            Pat::Ident(i) => self.add(&i.id.to_id()),
```

Comment:
I think this will mark `a` in `({ a = 'foo' })` as a binding because it's a pattern. Is this intended?

Reply:
It isn't intended, fixed.